[Collections] static analyzer fix and cleanup.

ianegordon opened the pull request:
Work in Progress.

Commit:
Collections static analyzer fix and cleanup.

Commit:
Collections static analyzer fix and cleanup.

Commit:
Collections static analyzer fix and cleanup.

ianegordon commented:
This is my initial pass on fixing the static analyzer and some general cleanup.

Please let me know if anything looks suspicious or non-standard.

willlarche commented on components/Collections/src/MDCCollectionViewController.m:
```diff
@@ -311,10 +334,10 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
-    [collectionView registerClass:[MDCCollectionInfoBarView class]
-        forSupplementaryViewOfKind:kind
-               withReuseIdentifier:identifier];
+    // KM: Simplify identifier
```
alignment

ianegordon replied:
All NSLogs (or any other lines with KM / KillMe) will be removed before merge.

ianegordon replied:
(The NSLogs will be useful for testing, so I'll keep them until everyone is satisfied)

ianegordon replied:
format_all run.

willlarche replied:
👌

willlarche commented on components/Collections/src/MDCCollectionViewController.m:
```diff
@@ -69,13 +85,20 @@ - (void)viewDidLoad {
+          forSupplementaryViewOfKind:footerKind
+                 withReuseIdentifier:classIdentifier];
+
+  NSLog(@"Registered Supplementary %@ : %@", headerKind, classIdentifier);  //KM
```
Is this important?

willlarche reviewed (changes requested):
Run format_all over this for a couple alignment things.

There's new nslogs. Were they needed?

Commit:
Ran scripts/format_all

Commit:
Ran scripts/format_all

Commit:
Ran scripts/format_all

willlarche commented:
Still needs @chriscox ' approval.

ianegordon committed:
[Typography] Workaround for medium system font on iOS 8 (#1292)

* [Typography] Workaround for medium system font on iOS 8

* Add issue number

ianegordon commented on components/Collections/examples/CollectionsEditingExample.m:
```diff
@@ -100,18 +103,36 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
+  //    return supplementaryView;
+  //  }
+
+  NSString *reuseIdentifier;
```
I am not sure of the best way to handle this.
An example should not have to deal with custom MDCCollection views.

Should we add a way to query your super class for a reuseIdentifier based on kind?
- (NSString *)reuseIdentifierForKind:(NSString *)kind

We cannot just invoke a super implementation as this method must return a nonnull value.

ShepJGoogle replied:
This will alter the use of the collection. In the MDCCollectionViewController class, within this method we also call updateControllerWithInfoBar:(MDCCollectionInfoBar *)infoBar which in short, assigns the supp view as a property to the controller so it can update it with each selection/deselection. So it's not just about updating the identifiers.

ShepJGoogle replied:
There's also some delegate methods for the MDCCollectionInfoBarViewDelegate protocol assigned to the info bars which the controller needs to be able to handle.

Commit:
Gigantic documentation update. (#1305)

There are more changes to come, but it's time we merged back.

Changes:
* Re-introduces the doc site scripts and files that had been deleted previously.
* Updates contributing docs to reflect doc structure changes.
* Applies common header structure across all component documentation.
* Fixes a bunch of liquid syntax bugs.
* Inlines Front Matter metadata in README.md files as HTML comments. Any file prefixed with

  <!--docs:
  ...yaml
  -->

  Will be converted into Front Matter when generating the site.
* .jekyll_prefix files, which previously stored the data, have been removed.

ianegordon committed:
Bump Podfiles.lock (#1304)

ShepJGoogle commented:
I think something to consider is decoupling our editing info bars from the collection view data source implementation and making them standalone views as part of the MDCCollectionViewController class. At a high level, the supplementary views we currently register as info header and info footer bars can be made as simple uiviews which we would insert as regular subviews in the view. It would remove all of the overheard and trickiness around having to inject the logic into the supplementary view data source method.

ianegordon commented:
Justin's comment about decoupling the info bars makes a lot of sense.
Are there any downsides?
@chriscox  ?

ianegordon committed:
[Dialogs] Fix AlertController rotation layout (#1306)

No longer uses outdated cached bounds values.

samnm committed:
[Feature highlight] Accessibility fixes (#1284)

* Allow highlighted view to be accessed with voiceover

* Double tap to dismiss voiceover hint

* NSLocalizedString

* Feature Highlight strings

chriscox commented:
@ShepJGoogle @ianegordon I do like Justin's idea about decoupling the inforbars.  Some of the requirements might be:

- What about creating a separate infobar view controller. One that handles both header/footer placement and visibility delegates. This controller is responsible for show/hide animations.
- Either `MDCCollectionView` or clients subclass would be required to trigger the infobar controller when entering edit more. I think if we handle in `MDCCollectionView`, then can be customized or overridden by client?
- The infobar background color, text color, and text could all be made customizable.
- Will need to provide delegates to forward touch events back to collection view controller.

chriscox committed:
Fixes catalog pod file ios-8 warnings. Closes #1313. (#1314)

ianegordon committed:
Minor Tweaks to address internal feedback. (#1312)

Commit:
Cleanup

Commit:
Cleanup

Commit:
Merge branch 'collectionsstaticanalyzer' of github.com:material-components/material-components-ios into collectionsstaticanalyzer

Commit:
Merge branch 'collectionsstaticanalyzer' of github.com:material-components/material-components-ios into collectionsstaticanalyzer

ianegordon commented:
I will open an issue to investigate changing our infobar handling.

I've created #1316 as a simplified PR to fix the analyzer warning.

ianegordon commented:
Infobar issue #1317
